### PR TITLE
Protocol Events

### LINKED ISSUE #298: Emit events when contracts are replaced (protocol replaceContract and loanToken setTarget)

### PR DESCRIPTION
**Pull request by smbsp**
Implementation Details: https://docs.google.com/document/d/1lA-8BNUck8L-bgiLTEUqWwBhSoceOheRQ4eW_q6Rcy4/edit
Closes #298

### CODE REVIEW

File: contracts/events/LoanOpeningsEvents.sol [hunk 1]
```diff
@@ -13,6 +13,8 @@ pragma solidity 0.5.17;
  * This contract contains the events for loan openings operations.
  * */
 contract LoanOpeningsEvents {
+	event ContractReplaced(address indexed changedBy, address indexed newTargetAddr, bytes32 indexed module);
```
Comment (tjcloa) [inline contracts/events/LoanOpeningsEvents.sol:16]: there is DRY design principle
repeating the same function is a signal to refactor
i suggest creating a ModulesCommonEvents.sol, place the function there contract and inherit from it to every module

Comment (tjcloa) [inline contracts/events/LoanOpeningsEvents.sol:16]:
```suggestion
	event ProtocolModuleContractReplaced(address indexed changedBy, address indexed newTargetAddr, bytes32 indexed module);
```
pls mind naming as it is used by FE and BE too

File: contracts/connectors/loantoken/LoanToken.sol [hunk 2]
```diff
@@ -34,6 +34,8 @@ contract LoanToken is AdvancedTokenStorage {
 	address internal target_;
 	address public admin;
 
+	event TargetSet(address indexed changedBy, address indexed newTargetAddr);
```
[the remark this answers is not in the document]
Comment (smbsp) [reply contracts/connectors/loantoken/LoanToken.sol]: Changes reverted

File: contracts/events/LoanOpeningsEvents.sol (hunk 1 above)
Comment (smbsp) [reply contracts/events/LoanOpeningsEvents.sol]: Done

Comment (smbsp) [reply contracts/events/LoanOpeningsEvents.sol]: Renamed and added prevModuleContractAddress

**Review by tjcloa — APPROVED** [review]

**Pull request merged**